core/search.go

```diff
@@ -30,56 +27,60 @@ func NewSearchRequests(r repo.Repo, cli *rpc.Client) *SearchRequests {
 
 	return &SearchRequests{
 		repo: r,
-		// node:  node,
-		cli: cli,
+		cli:  cli,
 	}
 }
 
+// Result struct
+type Result struct {
+	Type, ID string
+	Value    interface{}
+}
+
 // Search queries for items on qri related to given parameters
-func (d *SearchRequests) Search(p *repo.SearchParams, res *[]repo.DatasetRef) error {
+func (d *SearchRequests) Search(p *regclient.SearchParams, res *[]Result) error {
 	if d.cli != nil {
 		return d.cli.Call("SearchRequests.Search", p, res)
 	}
-	// if d.node != nil {
-	// 	r, err := d.node.Search(p.Query, p.Limit, p.Offset)
-	// 	if err != nil {
-	// 		return err
-	// 	}
 
-	if searchable, ok := d.repo.(repo.Searchable); ok {
-		results, err := searchable.Search(*p)
-		if err != nil {
-			log.Debug(err.Error())
-			return fmt.Errorf("error searching: %s", err.Error())
-		}
-		*res = results
-		return nil
+	reg := d.repo.Registry()
+
+	results, err := reg.Search(p)
+	if err != nil {
+		return err
 	}
 
-	return fmt.Errorf("this repo doesn't support search")
+	searchResults := make([]Result, len(results))
+	for i, result := range results {
+		searchResults[i].Type = result.Type
+		searchResults[i].ID = result.ID
+		searchResults[i].Value = result.Value
+	}
+	*res = searchResults
+	return nil
 }
 
 // ReindexSearchParams defines parmeters for
```

Comment:
Any particular reason this is being removed? If so we should delete it outright

Reply:
just postponed dealing with it to try to get this pr out sooner.  can file an issue or work on here depending on your preference